review/casetype.py

```
@@ -0,0 +1,128 @@
+    return any(k.lower() in text for k in keywords)
+
+
+def detect(case):
+    """Return {type, label, signals:{...}, rationale} for a case dict."""
+    tt = _titles_and_types(case)
+    court_cases = case.get("court_cases") or []
+
+    has_press_release = any(
+        (_any(t, _PRESS_RELEASE) and (_any(t, _CIAA) or st == "OFFICIAL_GOVERNMENT"))
+        or _any(t, _CIAA)
+        and _any(t, _PRESS_RELEASE)
```

Ruff (0.15.15)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)